attributes-kt/src/commonMain/kotlin/space/kscience/attributes/Attributes.kt

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018-2023 KMath contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package space.kscience.attributes
+
+/**
+ * A set of attributes. The implementation must guarantee that [content] keys correspond to their value types.
+ */
+public interface Attributes {
+    /**
+     * Raw content for this [Attributes]
+     */
+    public val content: Map<out Attribute<*>, Any?>
+
+    /**
+     * Attribute keys contained in this [Attributes]
+     */
+    public val keys: Set<Attribute<*>> get() = content.keys
+
+    /**
+     * Provide an attribute value. Return null if attribute is not present or if its value is null.
+     */
+    @Suppress("UNCHECKED_CAST")
+    public operator fun <T> get(attribute: Attribute<T>): T? = content[attribute] as? T
```

Comment:
I think it's a better idea to make `Attribute`'s type parameter inherit `Any`, so that `null` always means that there is no such attribute.

Reply:
OK. Then what about adding method like
```kotlin
    public operator fun <T> getMaybe(attribute: Attribute<T>): Option<T> = content[attribute] as? T
```
where (like in Rust)
```kotlin
sealed interface Option<out T>
data class Some<out T>(val value: T) : Option<T>
data object None: Option<Nothing>
```
that can be put into kmath-core.

Currently, I add similar methods in Kone as well.

Reply:
Yeah, I don't have any use case. I had some use cases for collections (and that's why I don't like Kotlin's and Java's maps). So I thought it would be good to have such a method that definitely won't mix together different situations.

So it's up to you to create it or not.